packages/web3-eth/src/eth_tx.ts

```diff
@@ -0,0 +1,315 @@
+			// transaction's number properties have been converted to HexStrings.
+			// JavaScript doesn't handle negative hex strings e.g. -0x1, but our
+			// numberToHex method does. -0x1 < 0 would result in false, so we must check if
+			// hex string is negative via the inclusion of -
+			transaction.gas === undefined ||
+			transaction.gasPrice === undefined ||
+			transaction.gas.startsWith('-') ||
+			transaction.gasPrice.startsWith('-')
+		)
+			throw new InvalidGasOrGasPrice({
+				gas: transaction.gas,
+				gasPrice: transaction.gasPrice,
+			});
+	} else if (
+		transaction.maxFeePerGas === undefined ||
+		transaction.maxPriorityFeePerGas === undefined ||
+		transaction.maxFeePerGas.startsWith('-') ||
+		transaction.maxPriorityFeePerGas.startsWith('-')
+	)
+		throw new InvalidMaxPriorityFeePerGasOrMaxFeePerGas({
+			maxPriorityFeePerGas: transaction.maxPriorityFeePerGas,
+			maxFeePerGas: transaction.maxFeePerGas,
+		});
+
+	const hasEip1559 =
+		transaction.maxFeePerGas !== undefined || transaction.maxPriorityFeePerGas !== undefined;
+	if (transaction.gasPrice !== undefined && (transaction.type === '0x2' || hasEip1559))
+		throw new Eip1559GasPriceError(transaction.gasPrice);
+	if ((transaction.type === '0x0' || transaction.type === '0x1') && hasEip1559)
+		throw new UnsupportedFeeMarketError({
+			maxFeePerGas: transaction.maxFeePerGas,
+			maxPriorityFeePerGas: transaction.maxPriorityFeePerGas,
+		});
+};
+
+export const validateTransactionForSigning = (
+	transaction: Transaction,
+	overrideMethod?: (transaction: Transaction) => void,
+) => {
+	if (overrideMethod !== undefined) {
+		overrideMethod(transaction);
+		return;
+	}
+
+	if (typeof transaction !== 'object' || transaction === null)
+		throw new InvalidTransactionObjectError(transaction);
+
+	const formattedTransaction = formatTransaction(transaction, ValidTypes.HexString);
+
+	validateCustomChainInfo(transaction);
+	validateChainInfo(transaction);
+	validateGas(formattedTransaction);
+
+	if (
+		formattedTransaction.nonce === undefined ||
+		formattedTransaction.chainId === undefined ||
+		formattedTransaction.nonce.startsWith('-') ||
+		formattedTransaction.chainId.startsWith('-')
+	)
+		throw new InvalidNonceOrChainIdError({
+			nonce: transaction.nonce,
+			chainId: transaction.chainId,
+		});
+};
+
+export async function populateTransaction<
+	DesiredType extends ValidTypes,
+	ReturnType = ValidReturnTypes[DesiredType],
+>(
+	transaction: Transaction,
+	web3Context: Web3Context<EthExecutionAPI>,
+	desiredType: DesiredType,
+	privateKey?: HexString,
+	overrideMethod?: (
+		transaction: Transaction,
+		web3Context: Web3Context<EthExecutionAPI>,
+	) => PopulatedUnsignedTransaction<ReturnType>,
+): Promise<PopulatedUnsignedTransaction<ReturnType>> {
+	if (overrideMethod !== undefined) return overrideMethod(transaction, web3Context);
+
+	const populatedTransaction = { ...transaction };
+	const web3Eth = new Web3Eth(web3Context.currentProvider);
```

Comment:
I think for scalability requirement, its more good not to create new `Web3Eth` object on every call inside this function.